cranelift/codegen/src/isa/x64/inst.isle

```diff
@@ -4298,9 +4298,18 @@
       (let ((size OperandSize (raw_operand_size_of_type ty)))
         (icmp_cond_result (x64_cmp size a b) (intcc_reverse cc))))
 
+;; Special case: use the test instruction for comparisons with 0.
+(rule 2 (emit_cmp cc a @ (value_type ty) (u64_from_iconst 0))
+      (let ((size OperandSize (raw_operand_size_of_type ty)))
+        (icmp_cond_result (x64_test size a a) cc)))
```

Comment:
To fix the issue on CI, I think this should work?

```
(let (
    (size OperandSize (raw_operand_size_of_type ty))
    (a Gpr (put_in_gpr a))
  )
  (icmp_cond_result (x64_test size a a) cc))
```

Basically by forcing `a` into a register earlier it's not separately converted to both `GprMem` and `Gpr`

Reply:
Yes, I think that's the right fix!